pkg/registry/core/shoot/strategy.go

```diff
@@ -143,6 +144,14 @@ func mustIncreaseGeneration(oldShoot, newShoot *core.Shoot) bool {
 	return false
 }
 
+func mustIncreaseGenerationForSpecChanges(oldShoot, newShoot *core.Shoot) bool {
+	if newShoot.Spec.Maintenance != nil && newShoot.Spec.Maintenance.ConfineSpecUpdateRollout != nil && *newShoot.Spec.Maintenance.ConfineSpecUpdateRollout {
+		return !apiequality.Semantic.DeepEqual(helper.HibernationIsEnabled(oldShoot), helper.HibernationIsEnabled(newShoot))
```

Comment:
Any special reason you use `DeepEqual`?

```suggestion
		return helper.HibernationIsEnabled(oldShoot) != helper.HibernationIsEnabled(newShoot)
```